paimon-core/src/main/java/org/apache/paimon/operation/metrics/CompactionMetrics.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.operation.metrics;
+
+import org.apache.paimon.annotation.VisibleForTesting;
+import org.apache.paimon.metrics.Histogram;
+import org.apache.paimon.metrics.MetricGroup;
+import org.apache.paimon.metrics.MetricRegistry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Metrics to measure a compaction. */
+public class CompactionMetrics {
+    private static final Logger LOG = LoggerFactory.getLogger(CompactionMetrics.class);
+    private static final int HISTOGRAM_WINDOW_SIZE = 10_000;
+    private static final String GROUP_NAME = "compaction";
+
+    private final MetricGroup metricGroup;
+
+    public CompactionMetrics(MetricRegistry registry, String tableName) {
+        this.metricGroup = registry.tableMetricGroup(GROUP_NAME, tableName);
```

Comment:
Why not use `bucketMetricGroup`? Compaction is a process happening in each bucket.